std/range/package.d

```diff
@@ -8146,433 +8306,824 @@ public:
+
+    // partial elements
+    assert(6.iota.slide!(Yes.withPartial)(3, 4).equal!equal(
+        [[0, 1, 2], [4, 5]]
+    ));
+    assert(6.iota.slide!(No.withPartial)(3, 4).equal!equal(
         [[0, 1, 2]]
     ));
-    assert(6.iota.slide(4, 1).equal!equal(
+
+    assert(6.iota.slide!(Yes.withPartial)(4, 1).equal!equal(
         [[0, 1, 2, 3], [1, 2, 3, 4], [2, 3, 4, 5]]
     ));
-    assert(6.iota.slide(4, 2).equal!equal(
+    assert(6.iota.slide!(Yes.withPartial)(4, 2).equal!equal(
         [[0, 1, 2, 3], [2, 3, 4, 5]]
     ));
-    assert(6.iota.slide(4, 3).equal!equal(
+
+    // partial elements
+    assert(6.iota.slide!(Yes.withPartial)(4, 3).equal!equal(
+        [[0, 1, 2, 3], [3, 4, 5]]
+    ));
+    assert(6.iota.slide!(No.withPartial)(4, 3).equal!equal(
         [[0, 1, 2, 3]]
     ));
-    assert(6.iota.slide(5, 1).equal!equal(
+
+    assert(6.iota.slide!(Yes.withPartial)(5, 1).equal!equal(
         [[0, 1, 2, 3, 4], [1, 2, 3, 4, 5]]
     ));
-    assert(6.iota.slide(5, 2).equal!equal(
+
+    // partial elements
+    assert(6.iota.slide!(Yes.withPartial)(5, 2).equal!equal(
+        [[0, 1, 2, 3, 4], [2, 3, 4, 5]]
+    ));
+    assert(6.iota.slide!(No.withPartial)(5, 2).equal!equal(
         [[0, 1, 2, 3, 4]]
     ));
-    assert(6.iota.slide(5, 3).equal!equal(
+
+    assert(6.iota.slide!(Yes.withPartial)(5, 3).equal!equal(
+        [[0, 1, 2, 3, 4], [3, 4, 5]]
+    ));
+    assert(6.iota.slide!(No.withPartial)(5, 3).equal!equal(
         [[0, 1, 2, 3, 4]]
     ));
 }
 
-// emptyness, copyability, strings
+// test emptyness, copyability and strings
 @safe pure nothrow unittest
 {
     import std.algorithm.comparison : equal;
     import std.algorithm.iteration : each, map;
 
     // check with empty input
     int[] d;
-    assert(d.slide(2).empty);
-    assert(d.slide(2, 2).empty);
+    assert(d.slide!(Yes.withPartial)(2).empty);
+    assert(d.slide!(Yes.withPartial)(2, 2).empty);
 
     // is copyable?
-    auto e = iota(5).slide(2);
+    auto e = iota(5).slide!(Yes.withPartial)(2);
     e.popFront;
     assert(e.save.equal!equal([[1, 2], [2, 3], [3, 4]]));
     assert(e.save.equal!equal([[1, 2], [2, 3], [3, 4]]));
     assert(e.map!"a.array".array == [[1, 2], [2, 3], [3, 4]]);
 
     // test with strings
     int[dstring] f;
-    "AGAGA"d.slide(3).each!(a => f[a]++);
+    "AGAGA"d.slide!(Yes.withPartial)(3).each!(a => f[a]++);
     assert(f == ["AGA"d: 2, "GAG"d: 1]);
 
     int[dstring] g;
-    "ABCDEFG"d.slide(3, 3).each!(a => g[a]++);
+    "ABCDEFG"d.slide!(Yes.withPartial)(3, 3).each!(a => g[a]++);
+    assert(g == ["ABC"d:1, "DEF"d:1, "G": 1]);
+    g = null;
+    "ABCDEFG"d.slide!(No.withPartial)(3, 3).each!(a => g[a]++);
     assert(g == ["ABC"d:1, "DEF"d:1]);
 }
 
-// test slicing, length
+// test length
+@safe pure nothrow unittest
+{
+    // Slides with fewer elements are empty or 1 for Yes.withPartial
+    static foreach (expectedLength, Partial; [No.withPartial, Yes.withPartial])
```

Comment:
👌

Reply:
BTW a first start for Phobos: https://github.com/dlang/phobos/pull/5989